src/balatro_gym/cards/joker/joker.py

```diff
@@ -0,0 +1,303 @@
+        return Rarity.COMMON
+
+    def get_chips_hand(
+        self, scored_cards: Sequence[PlayingCard], blind: BlindState, board: BoardState, scored_hand: PokerHandType
+    ) -> int:
+        return 50 if get_num_pairs(scored_cards) >= 1 else 0
+
+
+class WilyJoker(JokerBase):
+    _cost: int = 4
+
+    @property
+    def joker_type(self) -> Type:
+        return Type.CHIPS
+
+    @property
+    def rarity(self) -> Rarity:
+        return Rarity.COMMON
+
+    def get_chips_hand(
+        self, scored_cards: Sequence[PlayingCard], blind: BlindState, board: BoardState, scored_hand: PokerHandType
+    ) -> int:
+        return 100 if contains_three_set(scored_cards) else 0
+
+
+class CleverJoker(JokerBase):
+    _cost: int = 4
+
+    @property
+    def joker_type(self) -> Type:
+        return Type.CHIPS
+
+    @property
+    def rarity(self) -> Rarity:
+        return Rarity.COMMON
+
+    def get_chips_hand(
+        self, scored_cards: Sequence[PlayingCard], blind: BlindState, board: BoardState, scored_hand: PokerHandType
+    ) -> int:
+        return 80 if get_num_pairs(scored_cards) >= 2 else 0
+
+
+class DeviousJoker(JokerBase):
+    _cost: int = 4
+
+    @property
+    def joker_type(self) -> Type:
+        return Type.CHIPS
+
+    @property
+    def rarity(self) -> Rarity:
+        return Rarity.COMMON
+
+    def get_chips_hand(
+        self, scored_cards: Sequence[PlayingCard], blind: BlindState, board: BoardState, scored_hand: PokerHandType
+    ) -> int:
+        return 100 if len(get_straight(scored_cards, board)) == 5 else 0
+
+
+class CraftyJoker(JokerBase):
+    _cost: int = 4
+
+    @property
+    def joker_type(self) -> Type:
+        return Type.CHIPS
+
+    @property
+    def rarity(self) -> Rarity:
+        return Rarity.COMMON
+
+    def get_chips_hand(
+        self, scored_cards: Sequence[PlayingCard], blind: BlindState, board: BoardState, scored_hand: PokerHandType
+    ) -> int:
+        return 80 if len(get_flush(scored_cards, board)) > 0 else 0
+
+
+class HalfJoker(JokerBase):
+    _cost: int = 5
+
+    @property
+    def joker_type(self) -> Type:
+        return Type.ADDITIVE_MULT
+
+    @property
+    def rarity(self) -> Rarity:
+        return Rarity.COMMON
+
+    def get_mult(self, scored_cards: Sequence[PlayingCard], blind: BlindState, scored_hand: PokerHandType) -> int:
+        return 20 if len(scored_cards) <= 3 else 0
+
+
+class JokerStencil(JokerBase):
+    _cost: int = 8
+
+    @property
+    def joker_type(self) -> Type:
+        return Type.MULTIPLICATIVE
+
+    @property
+    def rarity(self) -> Rarity:
+        return Rarity.UNCOMMON
+
+    def get_multiplication(
+        self, scored_cards: Sequence[PlayingCard], blind: BlindState, board: BoardState, scored_hand: PokerHandType
+    ) -> float:
+        num_negative = sum([joker.edition.is_negative() for joker in board.jokers])
+        num_jokers = len(board.jokers)
+        return DEFAULT_JOKER_SLOTS - num_jokers + num_negative
```

Comment:
Similar to the `ConsumableState`, we should probably store the joker slots in the board state since this can be modified by the Antimatter voucher or the deck type (e.g, painted deck)